linkerd/app/outbound/src/http/logical.rs

```diff
@@ -14,101 +13,87 @@ struct ProfileRoute {
 }
 
 impl<N> Outbound<N> {
-    pub fn push_http_logical<NSvc>(self) -> Outbound<svc::ArcNewHttp<Logical>>
+    // TODO(ver) make the outer target type generic/parameterized.
+    pub fn push_http_logical<NSvc>(
+        self,
+    ) -> Outbound<
+        svc::ArcNewService<
+            Logical,
+            impl svc::Service<
+                    http::Request<http::BoxBody>,
+                    Response = http::Response<http::BoxBody>,
+                    Error = Error,
+                    Future = impl Send,
+                > + Clone,
+        >,
+    >
     where
         N: svc::NewService<Concrete, Service = NSvc> + Clone + Send + Sync + 'static,
-        NSvc: svc::Service<http::Request<http::BoxBody>, Response = http::Response<http::BoxBody>>
+        NSvc: svc::Service<
+                http::Request<http::BoxBody>,
+                Response = http::Response<http::BoxBody>,
+                Error = Error,
+            > + Clone
             + Send
+            + Sync
             + 'static,
-        NSvc::Error: Into<Error>,
         NSvc::Future: Send,
     {
         self.map_stack(|config, rt, concrete| {
-            let crate::Config {
-                discovery_idle_timeout,
-                http_request_buffer,
-                ..
-            } = config;
-
-            let split = concrete
-                .push_map_target(Concrete::from)
-                // This failfast is needed to ensure that a single unavailable
-                // balancer doesn't block the entire split.
-                //
-                // TODO(ver) remove this when we replace `split` with
-                // `distribute`.
-                .push_on_service(svc::FailFast::layer(http_request_buffer.failfast_timeout))
-                .check_new_service::<(ConcreteAddr, Logical), _>()
-                // Distribute requests over a distribution of balancers via a
-                // traffic split.
-                .push(profiles::split::layer())
-                .push_on_service(
+            // If there's no route, use the logical service directly; otherwise
+            // use the per-route stack.
+            concrete
+                .push(NewRouter::<_, _, Concrete>::layer(
                     svc::layers()
+                        // Maintain a per-route distributor over concrete
+                        // backends from the (above) concrete cache.
+                        .push(http::insert::NewInsert::<ProfileRoute, _>::layer())
```

Comment:
it's not _immediately_ obvious how the `NewInsert` layer is related to maintaining a per-route distributor, can we update the comment to explain that?